docs/question/html.md

```diff
@@ -1,13 +1,37 @@
-# Using Markdown in questions
+# `question.html`
 
-HTML and custom elements are great for flexibility and expressiveness. However, they're not great for working with large amounts of text, formatting text, and so on. [Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet) is a lightweight plaintext markup syntax that's ideal for authoring simple but rich text. To enable this, PrairieLearn adds a special `<markdown>` tag to questions. When a `<markdown>` block is encountered, its contents are converted to HTML. Here's an example `question.html` that utilizes this element:
+## Template file
+
+All `question.html` files are processed with the [Mustache template engine](https://mustache.github.io/mustache.5.html), which allows for basic dynamic rendering of HTML. For example, you can use the following syntax to conditionally render a piece of HTML:
+
+```html title="question.html"
+{{#params.show}}
+<p>This will only show up if params.show is true</p>
+{{/params.show}}
+```
+
+and this syntax to render over an array of strings (the `.` represents the current item in the array):
+
+```html title="question.html"
+{{#params.items}}
+<p>{{.}}</p>
+{{/params.items}}
+```
+
+!!! tip
+
+    If you use triple-braces (e.g. `{{{params.html}}}`) then raw HTML is substituted (don't use this unless you know you need it).
```

Comment:
I think it may be a good idea to strongly discourage the use of this feature with student submissions (pl-xss-safe may be suggested in that case).

Reply:
I suggested `pl-xss-safe` and explained the security context in 189fcab7d634a129d31bfb5c5fd1de4e2ee3cbd9.